File: productchallengeservice/src/main/java/br/com/joshua/productchallengeservice/entity/product/model/ProductModel.java
package br.com.joshua.productchallengeservice.entity.product.model;

import java.math.BigDecimal;

import br.com.joshua.productchallengeservice.entity.category.model.CategoryModel;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "products")
public class ProductModel {

	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String name;
	private String description;
	@ManyToOne
	@JoinColumn(name = "category_id")
	private CategoryModel categoryPath;
	private BigDecimal price;
	private Boolean available;

}

File: productchallengefront/src/app/pages/products/shared/product.model.ts
import { BaseResourceModel } from '../../../shared/models/base-resource.model';
import { Category } from './category.model';

export class Product extends BaseResourceModel {
  constructor(
    override id?: number,
    public name?: string,
    public description?: string,
    public categoryPath?: Category,
    public price?: number,
    public available?: boolean
  ) {
    super();
  }

  static fromJson(jsonData: any): Product {
    return Object.assign(new Product(), jsonData);
  }
}
